File: ticketdesk-ui/src/services/ticketService.js
import api from './api';

export const ticketService = {
  createTicket: async (ticketData) => {
    // ticketData: { title, description, category, priority }
    const response = await api.post('/tickets', ticketData);
    return response.data;
  },

  getTickets: async (params = {}) => {
    // params: { status, priority, category, page, size }
    const response = await api.get('/tickets', { params });
    return response.data;
  },

  getTicketById: async (id) => {
    const response = await api.get(`/tickets/${id}`);
    return response.data;
  },

  updateTicketStatus: async (id, statusData) => {
    // statusData: { status }
    const response = await api.patch(`/tickets/${id}/status`, statusData);
    return response.data;
  },

  assignTicket: async (id, agentId = null) => {
    const params = agentId ? { agentId } : {};
    const response = await api.patch(`/tickets/${id}/assign`, null, { params });
    return response.data;
  },

  addComment: async (ticketId, commentData) => {
    // commentData: { text }
    const response = await api.post(`/tickets/${ticketId}/comments`, commentData);
    return response.data;
  },

  getCommentsByTicketId: async (ticketId) => {
    const response = await api.get(`/tickets/${ticketId}/comments`);
    return response.data;
  },

  getDashboardSummary: async () => {
    const response = await api.get('/tickets/dashboard');
    return response.data;
  }
};

File: ticket-service/src/main/java/com/ticketdesk/ticketservice/service/TicketService.java
package com.ticketdesk.ticketservice.service;

import com.ticketdesk.ticketservice.dto.request.AddCommentRequest;
import com.ticketdesk.ticketservice.dto.request.CreateTicketRequest;
import com.ticketdesk.ticketservice.dto.request.UpdateTicketStatusRequest;
import com.ticketdesk.ticketservice.dto.response.CommentResponse;
import com.ticketdesk.ticketservice.dto.response.DashboardSummaryResponse;
import com.ticketdesk.ticketservice.dto.response.TicketResponse;
import com.ticketdesk.ticketservice.enums.TicketCategory;
import com.ticketdesk.ticketservice.enums.TicketPriority;
import com.ticketdesk.ticketservice.enums.TicketStatus;
import org.springframework.data.domain.Page;

import java.util.List;

public interface TicketService {

    TicketResponse createTicket(CreateTicketRequest request, Long currentUserId);

    Page<TicketResponse> getTickets(TicketStatus status, TicketPriority priority, TicketCategory category,
                                    String assignedStatus, int page, int size, Long currentUserId,
                                    boolean isAdmin, boolean isAgent);

    TicketResponse getTicketById(Long id, Long currentUserId, boolean isAgentOrAdmin);

    TicketResponse updateTicketStatus(Long id, UpdateTicketStatusRequest request, Long currentUserId, boolean isAgentOrAdmin);

    TicketResponse assignTicket(Long id, Long agentId, Long currentUserId, boolean isAgentOrAdmin);

    CommentResponse addComment(Long ticketId, AddCommentRequest request, Long currentUserId, boolean isAgentOrAdmin);

    List<CommentResponse> getCommentsByTicketId(Long ticketId, Long currentUserId, boolean isAgentOrAdmin);

    DashboardSummaryResponse getDashboardSummary();
}
